src/widgets/GitCloneForm.ts

```diff
@@ -16,23 +16,54 @@ export class GitCloneForm extends Widget {
   /**
    * Returns the input value.
    */
-  getValue(): string {
-    return encodeURIComponent(this.node.querySelector('input').value.trim());
+  getValue(): { url: string; versioning: boolean } {
+    return {
+      url: encodeURIComponent(
+        (
+          this.node.querySelector('#input-link') as HTMLInputElement
+        ).value.trim()
+      ),
+      versioning: Boolean(
+        encodeURIComponent(
+          (this.node.querySelector('#checkbox') as HTMLInputElement).checked
+        )
+      )
+    };
   }
 
   private static createFormNode(trans: TranslationBundle): HTMLElement {
     const node = document.createElement('div');
-    const label = document.createElement('label');
-    const input = document.createElement('input');
-    const text = document.createElement('span');
+    const inputWrapper = document.createElement('div');
+    const inputLinkLabel = document.createElement('label');
+    const inputLink = document.createElement('input');
+    const linkText = document.createElement('span');
+    const checkboxWrapper = document.createElement('div');
+    const checkboxLabel = document.createElement('label');
+    const checkbox = document.createElement('input');
 
-    node.className = 'jp-RedirectForm';
-    text.textContent = trans.__('Enter the Clone URI of the repository');
-    input.placeholder = 'https://host.com/org/repo.git';
+    node.className = 'jp-CredentialsBox';
+    inputWrapper.className = 'jp-RedirectForm';
+    checkboxWrapper.className = 'jp-CredentialsBox-wrapper';
+    checkboxLabel.className = 'jp-CredentialsBox-label-checkbox';
+    checkbox.id = 'checkbox';
+    inputLink.id = 'input-link';
+
+    linkText.textContent = trans.__('Enter the Clone URI of the repository');
+    inputLink.placeholder = 'https://host.com/org/repo.git';
+    checkboxLabel.textContent = 'Clone without metadata';
```

Comment:
The extension supports translation. For that you need to wrap the string to be translated with special `trans.__` function:

```suggestion
    checkboxLabel.textContent = trans.__('Clone without metadata');
```